school/diary/views.py

```diff
@@ -188,3 +249,34 @@ def achievements(request):
 def achievements_detail(request, achievement_id):
     reaching = get_object_or_404(Achievement, pk=achievement_id)
     return render(request, "diary/achievements_detail.html", {"achievement_key": reaching})
+
+
+# def add_achievement(request, football_player_id):
+#     # global player
+#     new_player = get_object_or_404(FootballPlayer, pk=football_player_id)
+#     form_data = AchievementForm(request.POST)
+#     if form_data.is_valid():
+#         Achievement.objects.create(football_player_achievements=new_player,
+#                                    tournament=form_data.cleaned_data['tournament'],
+#                                    achievement=form_data.cleaned_data['achievement'],
+#                                    scored_goals=form_data.cleaned_data['scored_goals'],
+#                                    appearances=form_data.cleaned_data['appearances'],
+#                                    clean_sheets=form_data.cleaned_data['clean_sheets'],)
+#
+#     return render(request, "diary/player.html", {"players": new_player, "error_message": form_data.errors})
+def add_achievement(request, football_player_id):
+    if request.POST:
+        if not player:
+            return redirect(login)
+        new_player = get_object_or_404(FootballPlayer, pk=football_player_id)
+    form_data = AchievementForm(request.POST)
+    if form_data.is_valid():
```

Comment:
Working with the form makes sense only when request.POST is True

Reply:
What does it mean?

Reply:
if request.POST: ?